### PR TITLE
ASimAuditEventBarracudaWAF.yaml-2

### PR DESCRIPTION
**Pull request by Alekhya0824**
Required items, please complete
   
  Change(s):
   - Created a new file for BaracudeCEF and added BaracudaCEF parser in it. In BaracudaWAF file removed BaracudaCEF parser and in ASimAuditEvent added BaracudeCEF Parser name  and same changes have done for vim as well.

   Reason for Change(s):
   - To Split the BaracudaWAF and BaracudaCEF parser separately

   Version Updated:
   - yes

   Testing Completed:
   - yes

   Checked that the validations are passing and have addressed any issues that are present:
   - yes

### CODE REVIEW

File: Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaWAF.yaml [hunk 1]
```diff
@@ -15,9 +15,9 @@ References:
 - Title: Logs type Description in Barracuda WAF
   Link: https://campus.barracuda.com/product/webapplicationfirewall/doc/78810008/types-of-logs/
 Description: |
-  This ASIM parser supports normalizing Barracuda WAF to the ASIM Audit Event schema.
-ParserName: ASimAuditEventBarracudaWAF
-EquivalentBuiltInParser: _ASim_AuditEvent_BarracudaWAF
+  This ASIM parser supports normalizing Barracuda WAF logs ingested in 'barracuda_CL' table to the ASIM Audit Event schema.
+ParserName: ASimAuditEventBarracudaCEF
+EquivalentBuiltInParser: _ASim_AuditEvent_BarracudaCEF
```
Comment (vakohl) [inline Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaWAF.yaml:20]: this is not CEF

File: Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaWAF.yaml [hunk 2]
```diff
@@ -15,9 +15,9 @@ References:
 - Title: Logs type Description in Barracuda WAF
   Link: https://campus.barracuda.com/product/webapplicationfirewall/doc/78810008/types-of-logs/
 Description: |
-  This ASIM parser supports normalizing Barracuda WAF to the ASIM Audit Event schema.
-ParserName: ASimAuditEventBarracudaWAF
-EquivalentBuiltInParser: _ASim_AuditEvent_BarracudaWAF
+  This ASIM parser supports normalizing Barracuda WAF logs ingested in 'barracuda_CL' table to the ASIM Audit Event schema.
+ParserName: ASimAuditEventBarracudaCEF
```
Comment (vakohl) [inline Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaWAF.yaml:19]: not a CEF parser

File: Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+- Title: ASIM Audit Event Schema
+  Link: https://aka.ms/ASimAuditEventDoc
+- Title: ASIM
+  Link: https://aka.ms/AboutASIM
+- Title: Logs type Description in Barracuda WAF
+  Link: https://campus.barracuda.com/product/webapplicationfirewall/doc/78810008/types-of-logs/
+Description: |
+  This ASIM parser supports normalizing Barracuda WAF logs ingested in 'CommonSecurityLog' table to the ASIM Audit Event schema.
+ParserName: ASimAuditEventBarracudaWAF
+EquivalentBuiltInParser: _ASim_AuditEvent_BarracudaWAF
```
Comment (vakohl) [inline Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml:20]: this one is CEF parser

File: Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+References:
+- Title: ASIM Audit Event Schema
+  Link: https://aka.ms/ASimAuditEventDoc
+- Title: ASIM
+  Link: https://aka.ms/AboutASIM
+- Title: Logs type Description in Barracuda WAF
+  Link: https://campus.barracuda.com/product/webapplicationfirewall/doc/78810008/types-of-logs/
+Description: |
+  This ASIM parser supports normalizing Barracuda WAF logs ingested in 'CommonSecurityLog' table to the ASIM Audit Event schema.
+ParserName: ASimAuditEventBarracudaWAF
```
Comment (vakohl) [inline Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml:19]: this one is CEF parser

File: Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+EquivalentBuiltInParser: _ASim_AuditEvent_BarracudaWAF
+ParserParams:
```
Comment (vakohl) [inline Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml:21]: can we map below?
(1) Warning: Missing recommended field [EventUid]

File: Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let barracudaSchema = datatable(
```
Comment (vakohl) [inline Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml:26]: we don't need to define schema for CEF parser

File: Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaWAF.yaml [hunk 7]
```diff
@@ -1,7 +1,7 @@
 Parser:
```
Comment (vakohl) [inline Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaWAF.yaml:1]: can we map below?
(1) Warning: Missing recommended field [EventUid]

Comment (vakohl) [reply Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaWAF.yaml:1]: can we check this?
(1) Warning: Empty value in 150 records (100.0%)  in mandatory field [EventEndTime] (Schema:AuditEvent)
(1) Warning: Empty value in 150 records (100.0%)  in mandatory field [EventStartTime] (Schema:AuditEvent)

File: Parsers/ASimAuditEvent/Parsers/vimAuditEventBarracudaCEF.yaml [hunk 8]
```diff
@@ -0,0 +1,232 @@
+Parser:
```
Comment (vakohl) [inline Parsers/ASimAuditEvent/Parsers/vimAuditEventBarracudaCEF.yaml:1]: (1) Warning: Missing recommended field [EventUid]

File: Parsers/ASimAuditEvent/Parsers/vimAuditEventBarracudaWAF.yaml [hunk 9]
```diff
@@ -1,7 +1,7 @@
 Parser:
```
Comment (vakohl) [inline Parsers/ASimAuditEvent/Parsers/vimAuditEventBarracudaWAF.yaml:1]: (1) Warning: Missing recommended field [EventUid]

File: Parsers/ASimAuditEvent/Parsers/vimAuditEventBarracudaWAF.yaml [hunk 10]
```diff
@@ -1,7 +1,7 @@
...
   Title: Audit Event ASIM parser for Barracuda WAF
```
Comment (vakohl) [inline Parsers/ASimAuditEvent/Parsers/vimAuditEventBarracudaWAF.yaml:2]: (1) Warning: Empty value in 150 records (100.0%)  in mandatory field [EventEndTime] (Schema:AuditEvent)
(1) Warning: Empty value in 150 records (100.0%)  in mandatory field [EventStartTime] (Schema:AuditEvent)

File: Parsers/ASimAuditEvent/Parsers/vimAuditEventBarracudaWAF.yaml [hunk 11]
```diff
@@ -192,103 +192,7 @@ ParserQuery: |
                 ManagementGroupName,
```
Comment (vakohl) [inline Parsers/ASimAuditEvent/Parsers/vimAuditEventBarracudaWAF.yaml:192]: please bring up the time filters, keep just after where not(disabled)

File: .gitignore [hunk 12]
```diff
@@ -335,4 +335,5 @@ ASALocalRun/
 Hunting Queries/DeployedQueries.json
```
Comment (vakohl) [inline .gitignore:335]: why changing this file?

**Review by vakohl — CHANGES_REQUESTED** [review]

File: Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaWAF.yaml (hunk 1 above)
Comment (Alekhya0824) [reply Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaWAF.yaml:20]: DONE

File: Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaWAF.yaml (hunk 2 above)
Comment (Alekhya0824) [reply Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaWAF.yaml:19]: DONE

File: Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml (hunk 3 above)
Comment (Alekhya0824) [reply Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml:20]: DONE

File: Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml (hunk 4 above)
Comment (Alekhya0824) [reply Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml:19]: DONE

File: Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml (hunk 6 above)
Comment (Alekhya0824) [reply Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml:26]: DONE

File: Parsers/ASimAuditEvent/Parsers/vimAuditEventBarracudaCEF.yaml (hunk 8 above)
Comment (Alekhya0824) [reply Parsers/ASimAuditEvent/Parsers/vimAuditEventBarracudaCEF.yaml:1]: done

File: Parsers/ASimAuditEvent/Parsers/vimAuditEventBarracudaWAF.yaml (hunk 9 above)
Comment (Alekhya0824) [reply Parsers/ASimAuditEvent/Parsers/vimAuditEventBarracudaWAF.yaml:1]: done

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Please act on reviewer's open comments. Thanks

File: Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml (hunk 5 above)
Comment (Alekhya0824) [reply Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml:21]: done

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
I can see a few open comments from the reviewer. Please act on them.
image
And also, one suggestion: once you're done with the open comment change, please click on the `Resolve Conversation` button.

File: Parsers/ASimAuditEvent/Parsers/vimAuditEventBarracudaWAF.yaml (hunk 11 above)
Comment (Alekhya0824) [reply Parsers/ASimAuditEvent/Parsers/vimAuditEventBarracudaWAF.yaml:192]: done

File: Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaWAF.yaml (hunk 7 above)
Comment (Alekhya0824) [reply Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaWAF.yaml:1]: done

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Did you respond to the open comment below? Thanks
image

File: Parsers/ASimAuditEvent/Parsers/vimAuditEventBarracudaWAF.yaml (hunk 10 above)
Comment (Alekhya0824) [reply Parsers/ASimAuditEvent/Parsers/vimAuditEventBarracudaWAF.yaml:2]: AS data is not present in workspace just closing this comment

File: Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml (hunk 5 above)
Comment (vakohl) [reply Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml:21]: missing in Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaWAF.yaml

Comment (vakohl) [reply Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml:21]: also in Parsers/ASimAuditEvent/Parsers/vimAuditEventBarracudaWAF.yaml

File: Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml (hunk 6 above)
Comment (vakohl) [reply Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml]: also need to remove from Parsers/ASimAuditEvent/Parsers/vimAuditEventBarracudaCEF.yaml

File: Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml (hunk 5 above)
Comment (Alekhya0824) [reply Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml:21]: I tried to adding EventUid in these WAF files but getting error 
in schema EventUid is not present
when i can schema file i'm not getting any error
image

File: Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml (hunk 6 above)
Comment (Alekhya0824) [reply Parsers/ASimAuditEvent/Parsers/ASimAuditEventBarracudaCEF.yaml]: done

**Review by vakohl — APPROVED** [review]

**Comment by Alekhya0824** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Comment by Alekhya0824** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: .gitignore (hunk 12 above)
Comment (Alekhya0824) [reply .gitignore]: to attach new schema file

**Review by vakohl — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**